Fix Data Retention & Image Upload in Workspace Settings

devansh-m12 opened the pull request:
This PR fixes two major issues in the Workspace Settings:

🖼️ Logo Upload Fix:

Fixed the issue where logo upload wasn’t working.

Root cause: Content Security Policy (CSP) was blocking image previews.

Fix: Updated the middilware image source to use blob: URLs, which are permitted under stricter CSP rules.

<img width="1455" height="834" alt="Screenshot 2025-07-11 at 12 44 43 AM" src="https://github.com/user-attachments/assets/be93d9ee-da06-48c9-8f73-0c07d80d6245" />


🎨 Brand Color & Website Retention Fix:

Previously, brandColor and website fields were not being retained after saving and refreshing.

Root cause: Initial form values were not being correctly populated from the persisted state.

Fix: Ensured the initial values are correctly loaded and reflected in the UI.


video: https://www.loom.com/share/3814b40cea8c416f82421e651990f9f9?sid=dfd51c7e-f907-45f4-afe0-d975af25427c

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved form behavior in the settings page by refining how default values are set for website and brand color fields.

* **Chores**
  * Updated security settings to allow images from blob URLs, enhancing compatibility for image sources.

* **Tests**
  * Added end-to-end tests for workspace settings, verifying form data display and update functionality.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

slavingia commented:
Please do not send me badly formed sentences with typos, that's grounds for being blocked.

slavingia commented:
Could you add e2e tests?

devansh-m12 commented:
I've added a test"s to check the rendered values and updated it accordingly.

<img width="767" height="272" alt="Screenshot 2025-07-11 at 8 09 53 PM" src="https://github.com/user-attachments/assets/7a415f5d-accf-4a46-a65b-8bdd2dbbfc0b" />

slavingia commented:
Test written here did not pass. Please be more careful next time
